app/dao/api_key_dao.py

```diff
@@ -56,21 +56,13 @@ def expire_api_key(service_id, api_key_id):
     db.session.add(api_key)
 
 
-# TODO: get rid of the sign_dangerously section once we've removed DANGEROUS_SALT and resigned the api keys
 def get_api_key_by_secret(secret):
     signed_with_all_keys = signer_api_key.sign_with_all_keys(str(secret))
     for signed_secret in signed_with_all_keys:
         try:
             return db.on_reader().query(ApiKey).filter_by(_secret=signed_secret).options(joinedload("service")).one()
         except NoResultFound:
             pass
-
-    signed_dangerous_with_all_keys = signer_api_key.sign_with_all_dangerously_salted_keys(str(secret))
-    for signed_secret in signed_dangerous_with_all_keys:
-        try:
-            return db.on_reader().query(ApiKey).filter_by(_secret=signed_secret).options(joinedload("service")).one()
-        except NoResultFound:
-            pass
     raise NoResultFound()
```

Comment:
Seems like we simplify the code now:
```
        return db.on_reader().query(ApiKey).filter_by(_secret=signed_secret).options(joinedload("service")).one()
```

and remove the try...except along with the manual raise at the end.

Reply:
Our SECRET_KEY is a list of 2 keys, and we have to support the api_key in the database being signed with either of them. (to continue working during the time between us changing the SECRET_KEY array to rotate a new key in and us running the resigning script).

So getting a `NoResultFound` from the first one is still ok, we could match the second time. If neither gets a match then we do have a legitimate `NoResultFound`.

(We should probably rename this to something like SIGNING_KEYS to make it less confusing)

Reply:
oh I missed the loop somehow 👍